MessageTrait.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Interop\Queue;
+
+/**
+ * The MessageTrait is a common implementation of the Message interface
+ */
+
+Trait MessageTrait
+{
+
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $body;
+
+    /**
+     * @var array
+     */
+    private $properties;
+
+    /**
+     * @var bool
+     */
+    private $redelivered;
+
+    /**
+     * @var array
+     */
+    private $headers;
+
+    public function setId(string $id = null): void
+    {
+        $this->id = $id;
+    }
+
+    public function getId(): ?string
+    {
+        return $this->id;
+    }
+
+    public function getBody(): string
+    {
+        return $this->body;
+    }
+
+    public function setBody(string $body): void
+    {
+        $this->body = $body;
+    }
+
+    public function setProperties(array $properties): void
+    {
+        $this->properties = $properties;
+    }
+
+    public function setProperty(string $name, $value): void
+    {
+        $this->properties[$name] = $value;
```

Comment:
if null value is passed the property (as well as header) should be unset, look at AmqpMessage implementation. https://github.com/queue-interop/amqp-interop/blob/a304a7a876ce872246e8fc01f6dc7b6c86c34277/src/Impl/AmqpMessage.php#L82-L86